### DISCUSSION TITLE
What is light channel used for?

### DISCUSSION THREAD

**Discussion by GerradLong** [google/filament#7998] [General]
I'm trying to understand what is the light channel: `LightsUib` `uint32_t channels; // 0x000c00ll (ll: light channels, c: caster)`

A part of this `uint` is `castShadow` info, sure. 

But I don't see any examples of 
`RenderableManager::setLightChannel`, 
`LightManager::setLightChannel` or 
`LightManager::Builder::lightChannel` 
functions being used.

and the shader code seems to check only with object flag channels to determine if the object should be shaded

```glsl
void evaluatePunctualLights(const MaterialInputs material,
        const PixelParams pixel, inout vec3 color) {
...

  int channels = object_uniforms_flagsChannels & 0xFF;

    // Iterate point lights
    for ( ; index < end; index++) {
        uint lightIndex = getLightIndex(index);
        Light light = getLight(lightIndex);
        if ((light.channels & channels) == 0) {
            continue;
        }

...

```

**Comment by romainguy** [comment]
Light channels can be used to have lights light only certain objects. For instance if you want a key light for a character that should apply only to that character and nothing else in the scene.
